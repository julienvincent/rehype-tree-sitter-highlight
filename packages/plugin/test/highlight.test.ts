import { expect, test } from "vitest";

import { fileURLToPath } from "node:url";
import path from "node:path";

import rehypeTreeSitter from "../src/index.ts";
import { rehype } from "rehype";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

test("highlights javascript code", () => {
  const html = `
<html>
<head></head>
<body>
  <pre>
    <code class="language-javascript">
      function sum(a, b) {
        console.log("this is a string");
        return a + b;
      }
    </code>
  </pre>
</body>
</html>`;

  const processor = rehype()
    .use(rehypeTreeSitter, {
      grammar_paths: [path.join(__dirname, "../../../fixtures/grammars/")],
    })
    .freeze();

  const output = processor.processSync(html).value;
  expect(output).matchSnapshot();
});

test("highlights markdown code with injections", () => {
  const html = `
<html>
<head></head>
<body>
  <pre>
    <code class="language-markdown">
      ## This is a title

      This is some text. Some text with \`a code block\`

      \`\`\`javascript
      console.log("And this is some javascript")
      \`\`\`

      Some more text
    </code>
  </pre>
</body>
</html>`;

  const processor = rehype()
    .use(rehypeTreeSitter, {
      grammar_paths: [path.join(__dirname, "../../../fixtures/grammars/")],
    })
    .freeze();

  const output = processor.processSync(html).value;
  expect(output).matchSnapshot();
});
