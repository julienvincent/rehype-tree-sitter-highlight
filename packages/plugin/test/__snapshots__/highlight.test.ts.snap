// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`highlights javascript code 1`] = `
"<html><head></head>
<body>
  <pre>    <code class="language-javascript"><span class="keyword">function</span><span> </span><span class="function">sum</span><span class="punctuation.bracket">(</span><span class="variable.parameter">a</span><span class="punctuation.delimiter">,</span><span> </span><span class="variable.parameter">b</span><span class="punctuation.bracket">)</span><span> </span><span class="punctuation.bracket">{</span><span>
  </span><span class="variable.builtin">console</span><span class="punctuation.delimiter">.</span><span class="function.method">log</span><span class="punctuation.bracket">(</span><span class="string">"this is a string"</span><span class="punctuation.bracket">)</span><span class="punctuation.delimiter">;</span><span>
  </span><span class="keyword">return</span><span> </span><span class="variable">a</span><span> </span><span class="operator">+</span><span> </span><span class="variable">b</span><span class="punctuation.delimiter">;</span><span>
</span><span class="punctuation.bracket">}</span></code>
  </pre>

</body></html>"
`;

exports[`highlights markdown code with injections 1`] = `
"<html><head></head>
<body>
  <pre>    <code class="language-markdown"><span class="punctuation.special">##</span><span> </span><span class="text.title">This is a title</span><span>

This is some text. Some text with </span><span class="punctuation.delimiter">\`</span><span class="text.literal">a code block</span><span class="punctuation.delimiter">\`</span><span>

</span><span class="punctuation.delimiter">\`\`\`</span><span class="text.literal">javascript
</span><span class="variable.builtin">console</span><span class="punctuation.delimiter">.</span><span class="function.method">log</span><span class="punctuation.bracket">(</span><span class="string">"And this is some javascript"</span><span class="punctuation.bracket">)</span><span>
</span><span class="punctuation.special"></span><span class="punctuation.delimiter">\`\`\`</span><span class="text.literal">
</span><span>
Some more tex</span></code>
  </pre>

</body></html>"
`;
